src/libical/icalcomponent.c

```diff
@@ -1406,6 +1406,12 @@ struct icaltimetype icalcomponent_get_dtstart(icalcomponent *comp)
 struct icaltimetype icalcomponent_get_dtend(icalcomponent *comp)
 {
     icalcomponent *inner = icalcomponent_get_inner(comp);
+    const icalcomponent_kind kind = icalcomponent_isa(inner);
+    if (kind != ICAL_VEVENT_COMPONENT && kind != ICAL_VAVAILABILITY_COMPONENT
```

Comment:
I think a switch() statement would be cleaner (and more extendible) than an if.  I need to tweak the VPOLL support to match the current IETF draft, and will have to add it in here eventually.